FirebaseCombineSwift/Sources/Auth/Auth+Combine.swift

```diff
@@ -0,0 +1,109 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if canImport(Combine) && swift(>=5.0) && canImport(FirebaseAuth)
+
+  import Combine
+  import FirebaseAuth
+
+  @available(swift 5.0)
+  @available(macOS 10.15, iOS 13, watchOS 6, tvOS 13, *)
+  public typealias AuthStateDidChangePublisher = AnyPublisher<(Auth, User?), Never>
+
+  @available(swift 5.0)
+  @available(macOS 10.15, iOS 13, watchOS 6, tvOS 13, *)
+  extension Auth {
+    // MARK: - Authentication State Management
+
+    /// Registers a publisher that publishes authentication state changes.
+    ///
+    /// The publisher emits values when:
+    ///
+    /// - It is registered,
+    /// - A user with a different UID from the current user has signed in, or
+    /// - The current user has signed out.
+    ///
+    /// - Returns: A publisher emitting (`Auth`, User`) tuples.
+    public func authStateDidChangePublisher() -> AuthStateDidChangePublisher {
+      let subject = PassthroughSubject<(Auth, User?), Never>()
+      let handle = addStateDidChangeListener { auth, user in
+        subject.send((auth, user))
+      }
+      return subject
+        .handleEvents(receiveCancel: {
+          self.removeStateDidChangeListener(handle)
+        })
+        .eraseToAnyPublisher()
+    }
+
+    // MARK: - Email/Password Authentication
+
+    /// Creates and, on success, signs in a user with the given email address and password.
+    /// - Parameters:
+    ///   - email: The user's email address.
+    ///   - password: The user's desired password.
+    /// - Returns: A publisher that emits the result of the sign in flow.
+    /// - Remark: Possible error codes:
+    /// - `AuthErrorCodeInvalidEmail` - Indicates the email address is malformed.
+    /// - `AuthErrorCodeEmailAlreadyInUse` - Indicates the email used to attempt sign up
+    ///   already exists. Call fetchProvidersForEmail to check which sign-in mechanisms the user
+    ///   used, and prompt the user to sign in with one of those.
+    /// - `AuthErrorCodeOperationNotAllowed` - Indicates that email and password accounts
+    ///   are not enabled. Enable them in the Auth section of the Firebase console.
+    /// - `AuthErrorCodeWeakPassword` - Indicates an attempt to set a password that is
+    ///   considered too weak. The NSLocalizedFailureReasonErrorKey field in the NSError.userInfo
+    ///   dictionary object will contain more detailed explanation that can be shown to the user.
+    /// - Remark: See `AuthErrors` for a list of error codes that are common to all API methods
+    public func createUser(withEmail email: String,
+                           password: String) -> Future<AuthDataResult, Error> {
+      Future<AuthDataResult, Error> { [weak self] promise in
```

Comment:
Would it not in general be ok to capture `self` strongly here? It of course depends on the implementation of the `Future` initializer, but even though the closure is escaping, it ought to be 'fire and forget'.
Alternatively, should the promise perhaps be called with an internal error in case `self` is nil? I think it may be expected that a `Future` eventually resolves.

Reply:
I'm no longer using `[weak self]` - see the initial comment on the PR for a rationale.